doc/neps/nep-0018-array-function-protocol.rst

```diff
@@ -799,6 +795,28 @@ If we want to include them in the future, the easiest way to do so would be to
 update the ``array_function_dispatch`` decorator to add them as function
 attributes.
 
+Protocols for methods
+~~~~~~~~~~~~~~~~~~~~~
+
+NumPy defines a number of classes with methods that act on NumPy arrays, e.g.,
+``vectorize.__call__`` and various ``random.RandomState`` methods. Such classes
```

Comment:
I don't think `__call__` is a good example - I only think the first of these is reasonable:

```
__array_function__(some_vectorize_obj, ...)
__array_function__(some_vectorize_obj.__call__, ...)
__array_function__(some_vectorize_obj.__call__.__call__, ...)
```

Reply:
Hmm, agree that the description can be seen as misleading. Certainly whatever is passed to `__array_function__` should be the "callable", i.e., the instance with a `__call__` method. In the context of the NEP, I guess the main question -- for which the answer is deferred -- is how to easily recognize those, as one cannot just, say, index a dict keyed with numpy functions with it, but will have to check the type or so.

Reply:
What if I replace `vectorize.__call__` with `vectorize` here?

Reply:
Well then it won't really be a method, so I don't think this heading will make sense.

Is the case you're actually trying to cover here "callable objects generated at runtime" (methods close over self, vectorize closes over a lambda, etc), for which there is no global name to check against?

Reply:
> In the context of the NEP, I guess the main question -- for which the answer is deferred -- is how to easily recognize those, as one cannot just, say, index a dict keyed with numpy functions with it, but will have to check the type or so.

This is a good way of putting it -- I'll add something about this